File: app/src/main/java/nz/co/chrisdrake/events/data/api/model/Event.java
package nz.co.chrisdrake.events.data.api.model;

import android.os.Parcelable;
import android.support.annotation.Nullable;
import com.google.auto.value.AutoValue;

@AutoValue public abstract class Event implements Parcelable {
    public abstract long id();
    public abstract String name();
    public abstract String url();
    @Nullable public abstract String thumbnailUrl();
    public abstract String description();
    public abstract String dateTimeSummary();
    public abstract String locationSummary();
    public abstract boolean hasMultipleSessions();

    public static Builder builder() {
        return new AutoValue_Event.Builder();
    }

    @AutoValue.Builder public abstract static class Builder {
        public abstract Builder id(long id);
        public abstract Builder name(String name);
        public abstract Builder url(String url);
        public abstract Builder thumbnailUrl(@Nullable String thumbnailUrl);
        public abstract Builder description(String description);
        public abstract Builder dateTimeSummary(String dateTimeSummary);
        public abstract Builder locationSummary(String locationSummary);
        public abstract Builder hasMultipleSessions(boolean hasMultipleSessions);
        public abstract Event build();
    }
}

File: app/src/main/java/nz/co/chrisdrake/events/domain/model/Event.kt
package nz.co.chrisdrake.events.domain.model

import android.os.Parcel
import android.os.Parcelable
import paperparcel.PaperParcel

@PaperParcel data class Event(
        @JvmField val id: Long,
        @JvmField val name: String,
        @JvmField val url: String,
        @JvmField val thumbnailUrl: String?,
        @JvmField val description: String,
        @JvmField val dateTimeSummary: String,
        @JvmField val locationSummary: String,
        @JvmField val hasMultipleSessions: Boolean
) : Parcelable {

    companion object {
        @JvmField val CREATOR = PaperParcelEvent.CREATOR
    }

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        PaperParcelEvent.writeToParcel(this, dest, flags)
    }
}
